cityofaustin/atd-moped
# Use relative timestamps in activity log

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/11356
- https://github.com/cityofaustin/atd-data-tech/issues/11357

## Testing
**URL to test:** local, but [netlify preview](https://deploy-preview-961--atd-moped-main.netlify.app/moped/projects) will be best


**Steps to test:**
1. Take a look at some existing projects and try to to find some activity events that match these criteria. You'll need to make your own edits to see "Just now" :)
 - less than 1 minute: `Just now`
 - less than 1 hour: `X minute(s) ago`
 - less than 1 day: `x hours ago`
 - less than 30 days: `x days ago`
 - else `m/dd/yyyy` - 1/23/2023

2. Hover over some of the relative timestamps in the activity log. A tooltip will open and show the full timestamp

<img width="207" alt="Screen Shot 2023-01-27 at 3 47 05 PM" src="https://user-images.githubusercontent.com/14793120/215193947-4d55d8c6-d590-4bd3-a277-0f97d34c73d1.png">

3. Run the tests. In the prompt that follows, you can use the `p` option to run tests by pattern then type `date` to select only our `dateAndTime` tests.

```shell
$ npm run test
```

You should see this test pass:

```shell
  formatRelativeDate()
    ✓ formats millesconds ago to a human description relative to now
```

---
#### Ship list
